### PR TITLE
Use local proxies instead of pypi-proxy.fly.dev for tests

### PR DESCRIPTION
**Pull request by zanieb**
The downside here is that we lose `https://` coverage, but I think it's worth it.

We've seen frequent test disruptions from fly.io being down, and it's always better for the test suite to be more self-contained. These test cases predated us having wiremock patterns in the suite, but now it's common.

### CODE REVIEW

File: crates/uv-auth/src/keyring.rs [hunk 1]
```diff
@@ -233,6 +233,26 @@ impl KeyringProvider {
                     Self::fetch_dummy(store, &host, username)
                 }
             };
+
+            // For non-HTTPS URLs, `store` includes the scheme in the service name
```
Comment (konstin) [inline crates/uv-auth/src/keyring.rs:237]: Is this a limitation of the dummy backend or of real backends too?

File: crates/uv/tests/it/auth.rs [hunk 2]
```diff
@@ -91,12 +92,23 @@ fn add_package_native_auth_realm() -> Result<()> {
...
 
     ----- stderr -----
-    Removed credentials for public@https://pypi-proxy.fly.dev/
+    Removed credentials for public@http://[LOCALHOST]/
     "
     );
 
+    // Reset the project so the previously-added `anyio` does not interfere with
+    // the resolution error message for `iniconfig`.
+    pyproject_toml.write_str(indoc::indoc! { r#"
```
Comment (konstin) [inline crates/uv/tests/it/auth.rs]: How do the proxy changes necessitate this?

**Review by konstin — APPROVED** [review]
One odd change, otherwise LGTM, this is a huge improvement!

File: crates/uv-auth/src/keyring.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-auth/src/keyring.rs:237]: This is a general limitation — it's intentional that we don't allow https credentials to bleed over to http. The lack of fallback here was an edge-case bug surfaced by switching the test cases to http.

File: crates/uv/tests/it/auth.rs (hunk 2 above)
Comment (zanieb) [reply crates/uv/tests/it/auth.rs]: For some reason, it introduced some non-determinism that affected the error messages. I didn't dig into why as it seemed harmless — but happy to do so.

Comment (zanieb) [reply crates/uv/tests/it/auth.rs]: Interesting, I had an agent look into the non-determinism

```                                                                
 The chain of events:                                                                                    
                                                                                                         
 1. uv add anyio succeeds → pyproject.toml now has anyio in dependencies, and anyio's Simple API         
 response is cached                                                                                      
 2. After logout, uv add iniconfig runs → modifies pyproject.toml to include both anyio and iniconfig,   
 then resolves                                                                                           
 3. The resolver fetches metadata for both packages from the auth-protected index                        
 4. For anyio: the cached response exists, but has freshness lifetime of Duration::ZERO because the      
 local mock proxy sets no cache headers (no Cache-Control, Expires, or Last-Modified). So it's           
 immediately stale → revalidation request → 401 Unauthorized → error (cached_client.rs line 532-540      
 returns the error instead of falling back to cache)                                                     
 5. For iniconfig: no cache, fresh request → 401 Unauthorized → error                                    
 6. Both packages are marked UnavailablePackage::NotFound                                                
 7. PubGrub's error message mentions whichever it encounters first — non-deterministic since both        
 requests complete near-simultaneously against the local server                                          
                                                                                                         
 Why this didn't happen with fly.dev: The nginx proxy forwarded PyPI's Last-Modified header, giving a    
 heuristic freshness lifetime of 10 minutes (line 970). The cached anyio response was still fresh →      
 served from cache without a network request → anyio resolved successfully → only iniconfig failed →     
 deterministic error.                                                                                    
                                                                                                         
 The fix (resetting pyproject.toml to remove anyio) is correct and minimal. An alternative would be to   
 add cache headers (e.g., Last-Modified) to the mock proxy's simple_api_response, but the pyproject      
 reset is simpler and avoids coupling the test to caching behavior.   
```


I might add forwarding of the cache headers?

Comment (konstin) [reply crates/uv/tests/it/auth.rs]: Afaik cache servers usually forward ~all headers (expect HOST and CONNECTION, https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/#passing-request-headers), should we do the same?

Comment (zanieb) [reply crates/uv/tests/it/auth.rs]: Only the file requests actually forward to PyPI, the simple API is generated locally so there are no headers for us to forward. Anyway, I just added some caching headers matching PyPI — that seems reasonable.
